src/compass/utils/validate_cslc.py

```diff
@@ -0,0 +1,142 @@
+                                     Validate reference and generated (secondary) S1 CSLC products""",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-r', '--ref-product', type=str, dest='ref_product',
+                        help='Reference CSLC product (i.e., golden dataset)')
+    parser.add_argument('-mr', '--ref-metadata', type=str, dest='ref_metadata',
+                        help='Reference CSLC metadata file (i.e., golden metadata)')
+    parser.add_argument('-s', '--sec-product', type=str, dest='sec_product',
+                        help='Secondary CSLC product to compare with reference')
+    parser.add_argument('-ms', '--sec-metadata', type=str, dest='sec_metadata',
+                        help='Secondary CSLC metadata to compare with reference metadata')
+    return parser.parse_args()
+
+
+def compare_cslc_products(file_ref, file_sec):
+    '''
+    Compare a reference and a newly generated
+    (i.e., secondary) CSLC product
+
+    Parameters
+    ----------
+    file1: str
+        File path to reference CSLC product (golden dataset)
+    file2: str
+        File path to generated CSLC product to use for comparison
+    '''
+
+    # Check if file paths exits
+    if not os.path.exists(file_ref):
+        print(f'ERROR reference CSLC product not found: {file_ref}')
+        return
+
+    if not os.path.exists(file_sec):
+        print(f'ERROR secondary CSLC product not found: {file_sec}')
+        return
+
+    # Extract some info from reference/secondary CSLC products
+    dataset_ref = gdal.Open(file_ref, gdal.GA_ReadOnly)
+    geotransform_ref = dataset_ref.GetGeoTransform()
+    nbands_ref = dataset_ref.RasterCount
+
+    dataset_sec = gdal.Open(file_sec, gdal.GA_ReadOnly)
+    geotransform_sec = dataset_sec.GetGeoTransform()
+    nbands_sec = dataset_sec.RasterCount
+
+    # Compare number of bands
+    print('Comparing CSLC number of bands ...')
+    if not nbands_ref == nbands_sec:
+        print(f'ERROR Number of bands in reference CSLC {nbands_ref} differs'
+              f'from number of bands {nbands_sec} in secondary CSLC')
+        return
+
+    print('Comparing geo transform arrays ...')
+    if not np.array_equal(geotransform_ref, geotransform_sec):
+        print(f'ERROR Reference geo transform array {dataset_ref} differs'
+              f'from secondary CSLC geo transform array {dataset_sec}')
+        return
+
+    # Compare amplitude of reference and generated CSLC products
+    slc_ref = dataset_ref.GetRasterBand(1).ReadAsArray()
+    slc_sec = dataset_sec.GetRasterBand(1).ReadAsArray()
+
+    diff_amp = np.abs(slc_ref) - np.abs(slc_sec)
+    diff_pha = np.angle(slc_ref * np.conj(slc_sec))
+
+    print('Check max amplitude difference between CSLC products is < 1.0e-12')
+    assert np.nanmax(diff_amp) < 1.0e-12
+    print('Check max phase difference between CSLC products is < 1.0e-12')
+    assert np.nanmax(diff_pha) < 1.0e-12
```

Comment:
My suggestion is to compare real and imag instead of phase and amp. 
Also relax the threshold to something around 1e-5 since we have single precision float data.

Reply:
Done.  A question, do we want to compare that the max difference of the real (imaginary) part or rather than the mean value is < 1e-5?